relays/ethereum/src/sync_loop.rs

```diff
@@ -682,4 +1049,134 @@ mod tests {
 		let (interval2, interval_after_reset) = run_backoff_test(Err(TestError(false)));
 		assert!(interval2 < interval_after_reset);
 	}
+
+	fn run_sync_loop_test(
+		best_source_header: TestHeader,
```

Comment:
Can we make all these parameters a struct? Will help with individual tests readabilty, now we pass a bunch of vecs and it's not clear what is what.